### PR TITLE
[android][test] Fix five tests that are failing on the community Android CI

### PR DESCRIPTION
**Pull request by finagolfin**
Update the Android doc with info about the latest LTS NDK not working.

@drodriguez, this should get the Android CI passing again.

### CODE REVIEW

File: test/Interop/Cxx/class/constructors-irgen-android.swift [hunk 1]
```diff
@@ -9,7 +9,7 @@ import Constructors
 import TypeClassification
 
 public func createHasVirtualBase() -> HasVirtualBase {
-  // ITANIUM_ARM: define protected swiftcc void @"$s7MySwift20createHasVirtualBaseSo0deF0VyF"(ptr noalias nocapture sret({{.*}}) %0)
+  // ITANIUM_ARM: define protected swiftcc void @"$s7MySwift20createHasVirtualBaseSo0deF0VyF"(ptr noalias sret({{.*}}) %0)
```
Comment (finagolfin) [inline test/Interop/Cxx/class/constructors-irgen-android.swift:12]: As done for many other tests in #68481

File: test/Interop/Cxx/templates/class-template-non-type-parameter-irgen.swift [hunk 2]
```diff
@@ -5,5 +5,5 @@ import ClassTemplateNonTypeParameter
 let p = MagicIntPair()
 let t = MagicIntTriple()
 
-// CHECK: @"${{s4main1pSo0042MagicArrayCInt_CUnsignedLong_2_DFABlHknrCcVvp|s4main1pSo0047MagicArrayCInt_CUnsignedLongLong_2_ofBJmlmartjdVvp}}"
-// CHECK: @"${{s4main1tSo0042MagicArrayCInt_CUnsignedLong_3_DFABlHknrCcVvp|s4main1tSo0047MagicArrayCInt_CUnsignedLongLong_3_ofBJmlmartjdVvp}}"
+// CHECK: @"${{s4main1pSo0042MagicArrayCInt_CUnsignedLong_2_DFABlHknrCcVvp|s4main1pSo0047MagicArrayCInt_CUnsignedLongLong_2_ofBJmlmartjdVvp|s4main1pSo0041MagicArrayCInt_CUnsignedInt_2_yDFGkvkBqzcVvp}}"
+// CHECK: @"${{s4main1tSo0042MagicArrayCInt_CUnsignedLong_3_DFABlHknrCcVvp|s4main1tSo0047MagicArrayCInt_CUnsignedLongLong_3_ofBJmlmartjdVvp|s4main1tSo0041MagicArrayCInt_CUnsignedInt_3_yDFGkvkBqzcVvp}}"
```
Comment (finagolfin) [inline test/Interop/Cxx/templates/class-template-non-type-parameter-irgen.swift:9]: This only fails on armv7, can simply disable it there instead if wanted.

File: test/SILOptimizer/character_literals.swift [hunk 3]
```diff
@@ -19,30 +19,33 @@ public func charArray(_ i: Int) -> [Character] {
 
 // NOTE: 97 = 'a'
 // NOTE: -2233785415175766016 = 0xE1 = 0xE0 (ASCII discrim) | 0x01 (count)
+// NOTE: On Android AArch64, this is shifted right by one byte: 0x00E1 = 63331869759897600
```
Comment (finagolfin) [inline test/SILOptimizer/character_literals.swift:22]: A direct consequence of #40779

**Comment by drodriguez** [comment]
@swift-ci please test

**Comment by finagolfin** [comment]
@drodriguez, passed CI, ready for review and merge.

**Comment by finagolfin** [comment]
Great, Android CI is green again.

**Pull request merged**